### ISSUE TITLE
GDM: The GNOME Keyring is not decrypted when logon with fingerprint

### ISSUE THREAD

**Issue by Icenowy** [AOSC-Dev/aosc-os-abbs#425]
Device: Lenovo Thinkpad X220i

After login with fingerprint, a dialog required my password is shown when starting Google Chrome (which made use of my keyring), and it said that my keyring is not decrypted.

Seems to be PAM configuration problem.

(GDM seems to use /etc/pam.d/gdm-fingerprint to login with fingerprint, not /etc/pam.d/gdm-password)

**Comment by MingcongBai** [comment]
Found two sources pointing to this issue.

- https://fedoraproject.org/wiki/Features/Fingerprint
- https://wiki.ubuntu.com/BluePrints/FingerprintAuth#Passwords_are_required_for_keyrings.2C_ecryptfs_to_work

Stating that:

"If you log in with a fingerprint, gnome-keyring cannot unlock the login keyring (since it uses the login password for that). Therefore, gnome-keyring will ask for your password the first time a secret is needed in the session. "

and

"You can't unlock a keyring or an encrypted filesystem without a password. Selectively disabling fingerprint auth for first login would be desirable and should be fairly easy, but isn't supported out of the box in the packages."

Respectively. This issue is largely "by-design" as it seems. I wouldn't encourage anyone to set their keyring password to blank, which is simply silly.

But the second point sounds quite interesting, if we could disable fingerprint login on first login, the issue could be hidden - but still a degrade in experience after all.

**Comment by MingcongBai** [comment]
> Lionium: So this is a #wontfix by upstream?

Yes, by at least one Red Hat bugzilla I saw.

**Comment by Icenowy** [comment]
@MingcongBai However, I think it can be fixed by us... and the fingerprint is enough to prove Icenowy is Icenowy...

**Comment by MingcongBai** [comment]
> @MingcongBai However, I think it can be fixed by us... and the fingerprint is enough to prove Icenowy is Icenowy...

Name an idea please?

**Comment by Icenowy** [comment]
@MingcongBai Just add session  optional  pam_gnome_keyring.so auto_start in /etc/pam.d/gdm-fingerprint

**Comment by MingcongBai** [comment]
> @MingcongBai Just add session optional pam_gnome_keyring.so auto_start in /etc/pam.d/gdm-fingerprint

Would you mind testing this fix on your machine?

**Comment by Icenowy** [comment]
@MingcongBai maybe...

**Comment by MingcongBai** [comment]
> @MingcongBai maybe...

Failed to work as stated by @Icenowy .

**Comment by MingcongBai** [comment]
Here's my speculation though... GDM fingerprint login really is not connected to the recorded password, while GNOME Keyring takes only password - no fingerprint unlock.

**Comment by biergaizi** [comment]
@Icenowy

I don't think this problem is fixable without damaging security in a wrong way. 

To *log in*, you provide your identity, which is called *authentication*; to read your encrypted data, you provide your passphrase, which is called *decryption*. These are two separate issues here, don't be confused.

The system doesn't care if you can read my encrypted emails, it still doesn't allow you to login. in the same way, mathematics doesn't care if you *are* the *owner* of the data, because even the system itself can not do anything without your key, whether you can *do the math* is the only thing matters. 

The only way to solve that is adding a backdoor, and talk to the system, *"Hey! I'm _____. Please use the key from a secret location in the filesystem"*, humm... sounds familiar?

TL; DR, you can not ask Gnome to factor RSA-4096 or brute-force AES by using one of your selfie photo! This issue should be close with *invalid*.
